Core/HLE/sceNetAdhoc.cpp

```diff
@@ -4705,3228 +4674,580 @@ int sceNetAdhocGetSocketAlert(int id, u32 flagPtr) {
-		return 0x80410005;
-	/*
-	if (Memory::Read_U32(netAdhocDiscoverBufAddr + 0xA4) == 0)
-		return 0x80411303; // Already Suspended?
-	if (Memory::Read_U32(netAdhocDiscoverBufAddr + 0x80) != 0)
-		return 0x80411303; // Already Suspended?
-	int ret = sceNetAdhocctl_lib_1572422C();
-	if (ret >= 0)
-		Memory::Write_U32(0, netAdhocDiscoverBufAddr + 0xA4);
-	return ret;
-	*/
-	// Since we don't know what this supposed to do, and we currently don't have a working AdhocDiscover yet, may be we should cancel the progress for now?
-	netAdhocDiscoverIsStopping = true;
-	return hleLogError(Log::sceNet, 0);
-}
-
-int sceNetAdhocDiscoverUpdate() {
-	DEBUG_LOG(Log::sceNet, "UNIMPL sceNetAdhocDiscoverUpdate() at %08x", currentMIPS->pc);
-	if (sceKernelCheckThreadStack() < 0x00000FF0)
-		return 0x80410005;
-
-	// TODO: Use switch case for each Step
-	if (netAdhocDiscoverStatus == NET_ADHOC_DISCOVER_STATUS_IN_PROGRESS) {
-		//u64 now = CoreTiming::GetGlobalTimeUsScaled();
-		if (netAdhocDiscoverIsStopping /*|| now >= netAdhocDiscoverStartTime + DISCOVER_DURATION_US*/) {
-			// Fake a successful completion after some time (or when detecting another player in the same Group?)
-			netAdhocDiscoverStatus = NET_ADHOC_DISCOVER_STATUS_COMPLETED;
-			if (netAdhocDiscoverParam)
-				netAdhocDiscoverParam->result = NET_ADHOC_DISCOVER_RESULT_CANCELED; // netAdhocDiscoverIsStopping ? NET_ADHOC_DISCOVER_RESULT_CANCELED : NET_ADHOC_DISCOVER_RESULT_PEER_FOUND;
-		}
-	}
-	return hleDelayResult(hleLogDebug(Log::sceNet, 0/*netAdhocDiscoverBufAddr*/), "adhoc discover update", 300); // FIXME: Based on JPCSP+prx, it seems to be returning a pointer to the internal buffer/struct (only when status = 1 ?), But when i stepped the code it returns 0 (might be a bug on JPCSP LLE Logging?)
-}
-
-const HLEFunction sceNetAdhocDiscover[] = {
-	{0X941B3877, &WrapI_U<sceNetAdhocDiscoverInitStart>,               "sceNetAdhocDiscoverInitStart",           'i', "x"        },
-	{0X52DE1B97, &WrapI_V<sceNetAdhocDiscoverUpdate>,                  "sceNetAdhocDiscoverUpdate",              'i', ""         },
-	{0X944DDBC6, &WrapI_V<sceNetAdhocDiscoverGetStatus>,               "sceNetAdhocDiscoverGetStatus",           'i', ""         },
-	{0XA2246614, &WrapI_V<sceNetAdhocDiscoverTerm>,                    "sceNetAdhocDiscoverTerm",                'i', ""         },
-	{0XF7D13214, &WrapI_V<sceNetAdhocDiscoverStop>,                    "sceNetAdhocDiscoverStop",                'i', ""         },
-	{0XA423A21B, &WrapI_V<sceNetAdhocDiscoverRequestSuspend>,          "sceNetAdhocDiscoverRequestSuspend",      'i', ""         },
-};
-
-void Register_sceNetAdhoc() {
-	RegisterModule("sceNetAdhoc", ARRAY_SIZE(sceNetAdhoc), sceNetAdhoc);
-	RegisterModule("sceNetAdhocMatching", ARRAY_SIZE(sceNetAdhocMatching), sceNetAdhocMatching);
-	RegisterModule("sceNetAdhocDiscover", ARRAY_SIZE(sceNetAdhocDiscover), sceNetAdhocDiscover);
-	RegisterModule("sceNetAdhocctl", ARRAY_SIZE(sceNetAdhocctl), sceNetAdhocctl);
```

Comment:
I think it would be better to split this into four functions, Register_sceNetAdhoc, Register_sceNetAdhocMatching etc. That way we could preserve the previous order. Now it becomes different since you only pulled out Matching.

Reply:
Does it matter though? It's not an `Init`/`Shutdown` routine, it's a simple `moduleDB.push_back(...)`.

Reply:
I think it does matter, it gives the modules different indices in the moduleDB vector, and I don't want to spend any time debugging any issues that might cause..  Module indices are used in our HLE syscall opcodes that the linker writes.